### PR TITLE
[xtext-eclipse#145] Extend implementations from core.

### PR DESCRIPTION
**Pull request by ArneDeutsch**
Change implementation to use new classesfrom xtext core plugin and
deprecate the UI classes.

Change-Id: I0d3faf0b96e967bf95c549c9e82749e95c24c1c1
Signed-off-by: Arne Deutsch <arne@idedeluxe.com>

### CODE REVIEW

**Comment by cdietrich** [comment]
hmmm no adaption of `AbstractCopyQualifiedNameHandler`
am not sure if thats good in connection with the `@Deprecated`

**Comment by cdietrich** [comment]
@szarnekow do we have a common pattern for such moves?

File: org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/copyqualifiedname/CopyQualifiedNameService.xtend [hunk 1]
```diff
@@ -12,9 +12,11 @@ import org.eclipse.emf.ecore.EObject
...
  * @author Anton Kosyakov - Initial contribution and API
  * @since 2.4
+ * @deprecated Use org.eclipse.xtext.naming.CopyQualifiedNameService instead
  */
-interface CopyQualifiedNameService {
+@Deprecated
+interface CopyQualifiedNameService extends org.eclipse.xtext.naming.CopyQualifiedNameService {
 
-	def abstract String getQualifiedName(EObject selectedElement, EObject context)
+	override abstract String getQualifiedName(EObject selectedElement, EObject context)
```
Comment (kthoms) [inline org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/copyqualifiedname/CopyQualifiedNameService.xtend]: remove method, it is already declared by superclass

File: org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/copyqualifiedname/CopyQualifiedNameService.xtend [hunk 2]
```diff
@@ -7,14 +7,12 @@
...
 /**
  * @author Anton Kosyakov - Initial contribution and API
  * @since 2.4
+ * @deprecated Use org.eclipse.xtext.naming.CopyQualifiedNameService instead
  */
-interface CopyQualifiedNameService {
-
-	def abstract String getQualifiedName(EObject selectedElement, EObject context)
+@Deprecated
+interface CopyQualifiedNameService extends org.eclipse.xtext.naming.CopyQualifiedNameService {
```
Comment (kthoms) [inline org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/copyqualifiedname/CopyQualifiedNameService.xtend]: It was decided that the interface will be prefixed with "I"

Comment (ArneDeutsch) [reply org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/copyqualifiedname/CopyQualifiedNameService.xtend]: Done.

**Pull request merged**